initrd/initrd.go

```diff
@@ -177,74 +177,73 @@ func NewFromMapping(workdir, output string, maps ...string) (*InitrdConfig, erro
 					return nil
 				}
 
-				internal = "." + strings.ReplaceAll(filepath.Join(initrdPath, internal), "//", "/")
+				if internal, err = filepath.Rel("/", internal); err != nil {
+					return fmt.Errorf("trimming leading separator from path %q: %w", internal, err)
+				}
 
 				info, err := d.Info()
 				if err != nil {
 					return err
 				}
 
-				if d.Type().IsDir() {
-					if err := writer.WriteHeader(&cpio.Header{
-						Name: internal,
-						Mode: cpio.TypeDir,
-					}); err != nil {
-						return err
+				header := &cpio.Header{
+					Name:    internal,
+					Mode:    cpio.FileMode(info.Mode().Perm()),
+					ModTime: info.ModTime(),
+				}
+
+				switch {
```

Comment:
I would also add a default case so we can see if people ever have any other type of files

Reply:
And what would it do?

Reply:
Just print something like `Unknown file type` and print it

Reply:
The only issue is that this method does not accept a `Context` so we cannot access the logger.
If I use `fmt.Print` it's going to mess the output up.

It is really critical to change the signature just to accept a logger? Any other file type would be irrelevant inside a unikernel.